packages/relay/src/ethers/signer.ts

```diff
@@ -41,8 +49,31 @@ export class DefenderRelaySigner extends Signer implements TypedDataSigner {
   ) {
     super();
     this.relayer = isRelayer(relayerCredentials) ? relayerCredentials : new Relayer(relayerCredentials);
-    if (options && options.speed && options.gasPrice) {
-      throw new Error(`Cannot set both speed and fixed gasPrice`);
+    if (options) {
+      const getUnnecesaryExtraFields = (invalidFields: (keyof GasOptions)[]) =>
+        invalidFields.map((field: keyof GasOptions) => options[field]).filter(Boolean);
+
+      if (options.speed) {
+        const unnecesaryExtraFields = getUnnecesaryExtraFields(['maxFeePerGas', 'maxPriorityFeePerGas', 'gasPrice']);
+
+        if (unnecesaryExtraFields.length > 0)
+          throw new Error(`Inconsistent options: speed + (${unnecesaryExtraFields}) not allowed`);
```

Comment:
I like this, it was a source of confusion

Reply:
Thanks! It's also on the backend if your some reason you'd like to give it a look